engine/src/main/java/org/hibernate/validator/internal/properties/javabean/JavaBeanExecutable.java

```diff
@@ -212,21 +194,43 @@ public String toString() {
 		);
 	}
 
-	private static Type getParameterGenericType(Type[] parameterTypes, Type[] genericParameterTypes, int parameterIndex) {
-		// getGenericParameterTypes() doesn't return synthetic parameters; in this case fall back to getParameterTypes()
-		Type[] typesToConsider;
-		if ( parameterIndex >= genericParameterTypes.length ) {
-			typesToConsider = parameterTypes;
-		}
-		else {
-			typesToConsider = genericParameterTypes;
+	private static List<JavaBeanParameter> getParameters(Executable executable) {
+		if ( executable.getParameterCount() == 0 ) {
+			return Collections.emptyList();
 		}
 
-		Type type = typesToConsider[parameterIndex];
+		List<JavaBeanParameter> parameters = new ArrayList<>( executable.getParameterCount() );
 
-		if ( type instanceof TypeVariable ) {
-			type = TypeHelper.getErasedType( type );
+		Parameter[] parameterArray = executable.getParameters();
+		Class<?>[] parameterTypes = executable.getParameterTypes();
+		// getGenericParameterTypes() does not include either the synthetic or the implicit parameters so we need to be
+		// extra careful
+		Type[] genericParameterTypes = executable.getGenericParameterTypes();
+
+		if ( parameterTypes.length == genericParameterTypes.length ) {
+			// this is the simple case where both arrays are consistent
+			// we could do without it but at some point, the behavior of getGenericParameterTypes() might be changed in
+			// Java and we'd better be ready.
```

Comment:
Well, the idea is that if the behavior is changed at some point, it will be that `genericParameterTypes` will contain all the parameters included the synthetic and implicit ones. In this case, it won't break anything.